File: software/examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_co2 import CO2

# Callback function for CO2 concentration greater than 750 ppm (parameter has unit ppm)
def cb_co2_concentration_reached(co2_concentration):
    print('CO2 Concentration: ' + str(co2_concentration) + ' ppm')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    co2 = CO2(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 10 seconds (10000ms)
    co2.set_debounce_period(10000)

    # Register threshold reached callback to function cb_co2_concentration_reached
    co2.register_callback(co2.CALLBACK_CO2_CONCENTRATION_REACHED, cb_co2_concentration_reached)

    # Configure threshold for "greater than 750 ppm" (unit is ppm)
    co2.set_co2_concentration_callback_threshold('>', 750, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var co2 = new Tinkerforge.BrickletCO2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        co2.setDebouncePeriod(10000);

        // Configure threshold for CO2 concentration "greater than 750 ppm" (unit is ppm)
        co2.setCO2ConcentrationCallbackThreshold('>', 750, 0);
    }
);

// Register CO2 concentration reached callback
co2.on(Tinkerforge.BrickletCO2.CALLBACK_CO2_CONCENTRATION_REACHED,
    // Callback function for CO2 concentration reached callback (parameter has unit ppm)
    function (co2Concentration) {
        console.log('CO2 Concentration: ' + co2Concentration + ' ppm');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
